### PR TITLE
Software Keymaps are not correctly interpreted / listened to.

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
This PR depends on https://github.com/patience-tema-baron/ember-keyboard/pull/87
Diff vs #87 
```
Comparing your current branch, key-combo-tests, to upgrade-dependencies
 addon-test-support/key-event.js                             | 17 +++++++++---
 addon-test-support/test-helpers.js                          |  4 +--
 addon/services/keyboard.js                                  |  5 ++++
 addon/utils/get-code.js                                     | 30 ++++++++++++++++++++-
 tests/acceptance/keyboard-combo-test.js                     | 66 ++++++++++++++++++++++++++++++++++++++++++++++
 tests/dummy/app/components/keyboard-press.js                | 25 ++++++++++++++++++
 tests/dummy/app/controllers/.gitkeep                        |  0
 tests/dummy/app/controllers/test-scenario/keyboard-combo.js | 19 +++++++++++++
 tests/dummy/app/router.js                                   |  1 +
 tests/dummy/app/templates/test-scenario/keyboard-combo.hbs  | 23 ++++++++++++++++
 10 files changed, 184 insertions(+), 6 deletions(-)
```

I think this'll resove #84, and superscede #85 as well as account for any other keyboard layout.

The gist is that on different keyboard layouts (esp those that aren't hardware enforced (like tho OS-language/region / keyboard layout settings), the KeyboardEvent`s `code` was reporting the hardware key.

Here is a test that demonstrates the event
```js
      test('keydown: Ctrl+k on a Dvorak Keyboard', async function(assert) {
        const ctrlKDownProperties = {
          code: 'KeyV',
          key: 'k',
          keyCode: 75,
          which: 75,
          ctrlKey: true
        };

        await textChanged(
          assert,
          () => triggerEvent(document.body, 'keydown', ctrlKDownProperties), {
            selectorName: 'ctrl-k',
            beforeValue: 'Ctrl+K not pressed',
            afterValue: 'Ctrl+K pressed'
          });
      });
```

This PR depends on #87, but the main fix is this:
```js
import codeMap from 'ember-keyboard/fixtures/code-map';

export default function getCode(event) {
  const { code, key, keyCode } = event;

  // Note that keyCode is deprecated
  // https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
  if (!code) {
    return codeMap[keyCode];
  }

  if (!code.includes('Key') || !key) {
    return code;
  }

  // If we have a software-applied key-remapping
  // For example:
  //   in Dvorak:
  //     pressing 'k'
  //       will give a code of 'KeyV'
  //       and a key of 'k'
  const codeLetter = code.charAt(code.length - 1);
  const keyboardLetter = codeLetter.toLowerCase();
  const typedLetter = key.toLowerCase();

  if (typedLetter === keyboardLetter) {
    return code;
  }

  const newCode = `Key${typedLetter.toUpperCase()}`;

  return newCode;
}
```

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
so, this actually still doesn't save the issue I'm having over at emberclear.

the problem is that ctrl+k just isn't even making it into the app code.

**Comment by NullVoxPopuli** [comment]
so, I guess this PR is _just_ a couple additional tests.

File: addon/services/keyboard.js [hunk 1]
```diff
@@ -68,6 +68,11 @@ export default Service.extend({
   },
 
   _respond(event) {
+    // if (!this.isPropagationEnabled) {
+    //   event.preventDefault();
+    //   event.stopPropagation();
+    // }
+
```
Comment (patience-tema-baron) [inline addon/services/keyboard.js]: A little dead code.

Comment (patience-tema-baron) [reply addon/services/keyboard.js]: Don't worry about this; you've done enough. I'll fix it during the release.

Comment (NullVoxPopuli) [reply addon/services/keyboard.js]: thanks a ton!!!

**Comment by NullVoxPopuli** [comment]
> My only concern about this approach is that some devs may want strict binding to a key, rather than to a code. For instance, if the dev wanted WASD to navigate through the app, regardless of key mappings.

yeah, I thought about that, too. Like, Blizzard games have their key-bindings by physical code from the keyboard, whereas slack, discord, vim, etc are all software key codes.

So... not sure how to support both styles. :-\

>  You've leveled up ember-keyboard in a pretty big way.

<3

**Comment by aabzhanova** [comment]
Hello friends!!  @NullVoxPopuli @patience-tema-baron  
This update broke my keyDown event  'KeyK+ctrl+alt' - it still works when I try 
'KeyK+ctrl' but any variation with 'alt' is not working. I'm on a mac/chrome

**Comment by NullVoxPopuli** [comment]
@aabzhanova do you have a reproduction?

**Comment by aabzhanova** [comment]
@NullVoxPopuli:
this doesn't fire
```on(keyPress('KeyK+ctrl+alt'), function() {
      //some action
    })```
this fires:
```on(keyPress('KeyK+ctrl'), function() {
      //some action
    })```
anything with `alt` doesn't fire

**Comment by NullVoxPopuli** [comment]
I mean, can you make a reproduction here: https://codesandbox.io/s/github/mike-north/ember-new-output (or in a standalone github repo)

it would help us out a ton <3

**Pull request merged**